hera_cal/lst_stack/metrics.py

```diff
@@ -0,0 +1,386 @@
+
+
+def _get_data_subset(
+    zscores: list[UVData],
+    bls: list[tuple[int, int, str]] | None = None,
+    band: tuple[int, int] | None = None,
+    nights='all',
+    pols='all',
+    selector=None
+):
+
+    allz = []
+
+    nbls = zscores[0].Nbls
+    datapols = utils.polnum2str(
+        zscores[0].polarization_array, x_orientation=zscores[0].x_orientation
+    )
+    datapairs = list(
+        zip(zscores[0].ant_1_array[:nbls], zscores[0].ant_2_array[:nbls])
+    )
+
+    if selector is not None:
+        allbls = [(a, b, p) for a, b in datapairs for p in datapols]
+        bls = get_selected_bls(allbls, min_days=0, selectors=selector)
+        selpols = {p for a, b, p in bls}
+        bls = list({bl[:2] for bl in bls})  # only antpairs
+
+        if 'ee' in selpols and 'nn' in selpols:
+            pols = 'all'
+        elif 'ee' in selpols:
+            pols = 'ee'
+        elif 'nn' in selpols:
+            pols = 'nn'
+
+    # Get pol indices
+    if pols == 'all':
+        pols = slice(None)
+    elif isinstance(pols, str):
+        pols = [datapols.index(pols)]
+    else:
+        pols = [datapols.index(p) for p in pols]
+
+    # Get bl indices
+    if bls is None:
+        bls = datapairs
+    if isinstance(bls, tuple) and len(bls) == 2:
+        bls = [datapairs.index(bls)]
+    elif isinstance(bls, list):
+        bls = [datapairs.index(bl) for bl in bls]
+
+    if band is None:
+        band = slice(None)
+    elif isinstance(band, tuple):
+        band = slice(band[0], band[1])
+
+    for lstbin, zscore in enumerate(zscores):
+        newshape = (zscore.Ntimes, zscore.Nbls, -1, zscore.Npols)
+        data = zscore.data_array[:, band].real.reshape(newshape)
+        flg = zscore.flag_array[:, band].reshape(newshape)
+
+        # make sure flagged stuff is nan
+        data[flg] *= np.nan
+        data = data[:, bls][..., pols]
+        # Get time indices
+        if nights != 'all':
+            data_nights = zscore.time_array[::zscore.Nbls].astype(int).tolist()
+
+            if isinstance(nights, int):
+                data = data[[data_nights.index(nights)]]
+            else:
+                data = data[[data_nights.index(n) for n in nights]]
+
+        allz.append(data)
+    return allz
+
+
+def get_data_subset(
+    data: list[UVData], bls=None, band=None, nights='all', pols='all', selector=None
+):
+    """
+    Get a subset of data from a list of UVData objects.
+
+    Returns a single flat array with all the datapoints that satisfy the subset criteria.
+    """
+    subsets = _get_data_subset(zscores=data, bls=bls, band=band, nights=nights, pols=pols, selector=selector)
+    allz = []
+    for subset in subsets:
+        subset = subset.flatten()
+        subset = subset[~np.isnan(subset)]
+        allz.append(subset)
+    return np.concatenate(allz)
+
+
+def get_data_subset_mean(
+    data: list[UVData], bls=None, band=None, nights='all', pols='all', mean_over=None, selector=None
```

Comment:
Could you make it a little more clear what `selector` is doing here? From reading `get_selected_bls`, it seems like it provides a way for users to filter out specific baselines based on some criteria by defining a function or list of functions. It would be nice to have some documentation of that parameter here or somewhere else in this file.